BSP/Inc/BSP_SPI.h

```diff
@@ -25,7 +25,7 @@ typedef enum {
  * @param   spi_os pointer to struct that holds the SPI specific pend()/post() functions 
  * @return  None
  */
-void BSP_SPI_Init(spi_port_t port, bsp_os_t *spi_os);
+void BSP_SPI_Init(spi_port_t port, bsp_os_t *spi_os, bool baremetal);
```

Comment:
Has the bare-metal code been tested? Do we want to keep it if we can do without it for now or are we saving it for another case?